src/EditorFeatures/Core/InheritanceMargin/InheritanceMarginDefinitionItemExtensions.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Editor.FindUsages;
+using Microsoft.CodeAnalysis.FindUsages;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+
+namespace Microsoft.CodeAnalysis.Editor.InheritanceMargin
+{
+    internal static class InheritanceMarginDefinitionItemExtensions
+    {
+        /// <summary>
+        /// Create the DefinitionItem based on the numbers of locations for <paramref name="symbol"/>.
+        /// If there is only one location, create the DefinitionItem contains only the documentSpan or symbolKey to save memory.
+        /// Because in such case, when clicking InheritanceMarginGlpph, it will directly navigate to the symbol.
+        /// Otherwise, create the full non-classified DefinitionItem. Because in such case we want to display all the locations to the user
+        /// by reusing the FAR window.
+        /// </summary>
+        public static async Task<DefinitionItem> ToSlimDefinitionItemAsync(this ISymbol symbol, Solution solution, CancellationToken cancellationToken)
+        {
+            var locations = symbol.Locations;
+            if (locations.IsEmpty)
+            {
+                return DefinitionItem.CreateNonNavigableItem(
+                    tags: GlyphTags.GetTags(symbol.GetGlyph()),
+                    displayParts: FindUsagesHelpers.GetDisplayParts(symbol));
+            }
+
+            if (locations.Length > 1)
+            {
+                return await symbol.ToNonClassifiedDefinitionItemAsync(
+                    solution,
+                    includeHiddenLocations: false,
+                    cancellationToken: cancellationToken).ConfigureAwait(false);
+            }
+
+            var location = locations[0];
+            if (location.IsInMetadata)
+            {
+                return DefinitionItem.CreateMetadataDefinition(
+                    ImmutableArray<string>.Empty,
+                    ImmutableArray<TaggedText>.Empty,
+                    ImmutableArray<TaggedText>.Empty,
+                    solution,
+                    symbol);
+            }
+            else if (location.IsInSource && location.IsVisibleSourceLocation())
+            {
+                var document = solution.GetDocument(location.SourceTree);
```

Comment:
❔ Does this work with source generator outputs? If not, we need to determine whether this _used_ to work with source generated outputs (thus a regression) or _never_ did (thus a request for a new feature). Either way we would want a follow-up bug and/or tests.

Reply:
Verified on compiler's solution 
![image](https://user-images.githubusercontent.com/24360909/150013641-c2fb9dbd-2224-4ee5-844b-42342e912b04.png)
I can navigate to another source generator output 
![image](https://user-images.githubusercontent.com/24360909/150013661-47a0f3e6-42bf-4735-912f-d3d006df9df1.png)